flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/CodeGenUtils.scala

```diff
@@ -312,6 +310,17 @@ object CodeGenUtils {
     case _ => "null"
   }
 
+  /**
+   * Gets a properly typed default value for a primitive type. For example: ((short) -1) for
+   * SMALLINT, ((byte) -1) for TINYINT. This ensures the generated code compiles correctly when the
+   * default value is used in method calls expecting specific primitive types.
+   */
+  def primitiveDefaultValueWithCast(logicalType: LogicalType): String = {
+    val resultTerm = primitiveDefaultValue(logicalType)
+    val resultTypeTerm = primitiveTypeTermForType(logicalType)
+    s"(($resultTypeTerm) $resultTerm)"
+  }
```

Comment:
isn't this code becoming more error prone since now need to remember that `primitiveDefaultValue` doesn't work properly and need to call extra cast separately?

Also `primitiveDefaultValue` doesn't seem following scaladoc ...